db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/WindowResultSetNode.java

```diff
@@ -495,6 +495,35 @@ private void addOverColumns(Map<String, ColumnRefPosition> pulledUp) throws Stan
                         // Cache the ref
                         pulledUp.put(expKey, new ColumnRefPosition(bottomRC, childRCL.size()));
                     }
+                } else {
+                    // Check again if we find match, if not then add it in cache ref if we have the equivalent match
+                    if (pulledUp.get(expKey) == null) {
```

Comment:
I don't understand. To get here in line 444 "(pulledUp.get(expKey) == null)" evaluated to true, then in line 445 "(! contains(expRC, winRCL))" evaluated to false. Now we are evaluating  "(pulledUp.get(expKey) == null)" again, did either of pulledUp or expKey change in the meantime?